frontend/src/app/modules/album/components/create-edit-album/create-edit-album/create-edit-album.component.ts

```diff
@@ -223,4 +229,33 @@ export class CreateEditAlbumComponent implements OnInit, OnDestroy {
         });
     }
   };
+
+  setPublicStatus() {
+    let albumPublicStatus: AlbumPublicStatus;
+
+    if (this.album.isPublished) {
+      albumPublicStatus = {
+        ...this.album,
+        isPublished: false
+      };
+    }
+    else {
+      albumPublicStatus = {
+        ...this.album,
+        isPublished: true
+      };
+    }
+
+    this._albumService.changeAlbumPublicStatus(albumPublicStatus)
+      .subscribe({
+        next: (data) => {
+          this.album = {
+            ...this.album,
+            isPublished: data.isPublished
```

Comment:
You already changed the status before sending the request, there's no need to reassign the value. Especially since you don't change it in backend side